functions/ocr/app/requirements.txt

```
@@ -1,5 +1,17 @@
-google-cloud-pubsub==2.9.0
+# Explicitly listing more packages than just the direct script dependencies to
+# avoid PIP taking too long while resolving dependencies.
+# See https://pip.pypa.io/en/stable/topics/dependency-resolution/#backtracking
+google-api-core==2.11.0
+google-auth==2.16.2
+google-cloud-core==2.3.2
+google-cloud-pubsub==2.15.1
 google-cloud-storage==2.0.0; python_version < '3.7'
-google-cloud-storage==2.1.0; python_version > '3.6'
-google-cloud-translate==3.8.0
-google-cloud-vision==3.1.0
+google-cloud-storage==2.7.0; python_version >= '3.7'
```

3.6 and below are not used for functions (2.7 is ignored, 3.6 tests are not used) so this can edited with these two suggestions:

```suggestion
google-cloud-storage==2.7.0
```